### PR TITLE
[Fix] fix a bug when module is missing in low version of bitsandbytes

### PR DESCRIPTION
**Pull request by Ben-Louis**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers. By the way, if you're not familiar with how to use pre-commit to fix lint issues or add unit tests, please refer to [Contributing to OpenMMLab](https://mmengine.readthedocs.io/en/latest/notes/contributing.html).

## Motivation

In [`register_bitsandbytes_optimizers`](https://github.com/open-mmlab/mmengine/blob/6c5eebb823e3c9381d63fd0cd1873ed1bd9ee9de/mmengine/optim/optimizer/builder.py#L132C14-L154), some optimizer modules are missing from a low version of `bitsandbytes`, resulting in a bug for users with lower versions. (e.g. https://github.com/open-mmlab/mmpose/issues/2747)

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMPretrain.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmengine/optim/optimizer/builder.py [hunk 1]
```diff
@@ -146,7 +146,7 @@ def register_bitsandbytes_optimizers() -> List[str]:
                 'PagedAdamW8bit', 'LAMB8bit', 'LARS8bit', 'RMSprop8bit',
                 'Lion8bit', 'PagedLion8bit', 'SGD8bit'
         ]:
-            _optim = getattr(bnb.optim, module_name)
```
Comment (HAOCHENYE) [inline mmengine/optim/optimizer/builder.py:149]: Register a None optimizer seems weird, maybe we should skip registering the optimizer if it does not exist

Comment (Ben-Louis) [reply mmengine/optim/optimizer/builder.py:149]: If `_optim` is `None`, the result of `inspect.isclass(_optim)` is `False` and the registering will be skipped
https://github.com/open-mmlab/mmengine/blob/2135d3703ebffad6097c78cc4ce04aac2124360d/mmengine/optim/optimizer/builder.py#L149-L153

Comment (HAOCHENYE) [reply mmengine/optim/optimizer/builder.py:149]: From the perspective of code readability, explicitly skipping registration is preferable to implicitly skipping it by passing in `None`.

BTW, I'm not familiar with the optimizer in bitsandbytes, why don't we refer to the implementation here

https://github.com/open-mmlab/mmengine/blob/2135d3703ebffad6097c78cc4ce04aac2124360d/mmengine/optim/optimizer/builder.py#L120

 to register the optimizer?

Comment (Ben-Louis) [reply mmengine/optim/optimizer/builder.py:149]: Me neither. I just attempted to fix the bug with minimal changes. I'm uncertain if modifying the builder function for `bitsandbytes` to resemble `Sophia`'s could cause any issues

**Review by zhouzaida — APPROVED** [review]

**Pull request merged**